concordium-consensus/tests/scheduler/SchedulerTests/SmartContracts/V1/Iterator.hs

```diff
@@ -44,82 +50,61 @@ iteratorSourceFile = "./testdata/contracts/v1/iterator.wasm"
-                      metadata = makeDummyHeader alesAccount 2 100000,
-                      keys = [(0, [(0, alesKP)])]
+                      metadata = makeDummyHeader accountAddress0 2 100000,
+                      keys = [(0, [(0, keyPair0)])]
                     },
-                  (SuccessWithSummary initializationCostCheck, iteratorSpec)
-                ),
-                ( TJSON
+              taaAssertion = \result _ ->
+                return $ Helpers.assertSuccess result
+            },
+          Helpers.TransactionAndAssertion
+            { taaTransaction =
+                TJSON
                     { payload = Update 0 (Types.ContractAddress 0 0) "iterator.iteratetest" BSS.empty,
-                      metadata = makeDummyHeader alesAccount 3 100000,
-                      keys = [(0, [(0, alesKP)])]
+                      metadata = makeDummyHeader accountAddress0 3 100000,
+                      keys = [(0, [(0, keyPair0)])]
                     },
-                  (SuccessWithSummary ensureSuccess, iteratorSpec)
-                ),
-                ( TJSON
+              taaAssertion = \result _ ->
+                return $ Helpers.assertSuccess result
+            },
+          Helpers.TransactionAndAssertion
+            { taaTransaction =
+                TJSON
                     { payload = Update 0 (Types.ContractAddress 0 0) "iterator.lockingtest" BSS.empty,
-                      metadata = makeDummyHeader alesAccount 4 100000,
-                      keys = [(0, [(0, alesKP)])]
+                      metadata = makeDummyHeader accountAddress0 4 100000,
+                      keys = [(0, [(0, keyPair0)])]
                     },
-                  (SuccessWithSummary ensureSuccess, iteratorSpec)
-                )
-            ]
-        }
-    ]
-  where
-    iteratorSpec _ = return ()
-    deploymentCostCheck :: TVer.BlockItemWithStatus -> Types.TransactionSummary -> Expectation
-    deploymentCostCheck _ Types.TransactionSummary{..} = do
-        checkSuccess "Module deployment failed: " tsResult
-        moduleSource <- BS.readFile iteratorSourceFile
-        let len = fromIntegral $ BS.length moduleSource
-            -- size of the module deploy payload
-            payloadSize = Types.payloadSize (Types.encodePayload (Types.DeployModule (WasmModuleV0 (WasmModuleV ModuleSource{..}))))
-            -- size of the transaction minus the signatures.
-            txSize = Types.transactionHeaderSize + fromIntegral payloadSize
-        -- transaction is signed with 1 signature
-        assertEqual "Deployment has correct cost " (Cost.baseCost txSize 1 + Cost.deployModuleCost len) tsEnergyCost
-    -- This only checks that the cost of initialization is correct.
-    -- If the state was not set up correctly the latter tests in the suite will fail.
-    initializationCostCheck :: TVer.BlockItemWithStatus -> Types.TransactionSummary -> Expectation
-    initializationCostCheck _ Types.TransactionSummary{..} = do
```

Comment:
In my opinion it is better to have a dedicated unit test for the init cost and deployment cost instead of checking this everywhere.
But I have added these checks again everywhere, and so we can leave that discussion for another time.